contrib/offline/manage-offline-container-images.sh

```diff
@@ -18,89 +97,107 @@ function create_container_image_tar() {
-	rm -rf ${IMAGE_DIR}
-	mkdir  ${IMAGE_DIR}
-	cd     ${IMAGE_DIR}
+	rm -f  "${IMAGE_TAR_FILE}"
+	rm -rf "${IMAGE_DIR}"
+	mkdir  "${IMAGE_DIR}"
+	cd     "${IMAGE_DIR}"
 
 	sudo docker pull registry:latest
 	sudo docker save -o registry-latest.tar registry:latest
 
 	for image in ${IMAGES}
 	do
-		FILE_NAME="$(echo ${image} | sed s@"/"@"-"@g | sed s/":"/"-"/g)".tar
-		set +e
-		for step in $(seq 1 ${RETRY_COUNT})
-		do
-			sudo docker pull ${image}
-			if [ $? -eq 0 ]; then
-				break
-			fi
-			echo "Failed to pull ${image} at step ${step}"
-			if [ ${step} -eq ${RETRY_COUNT} ]; then
-				exit 1
-			fi
-		done
-		set -e
-		sudo docker save -o ${FILE_NAME}  ${image}
-
-		# NOTE: Here removes the following repo parts from each image
-		# so that these parts will be replaced with Kubespray.
-		# - kube_image_repo: "registry.k8s.io"
-		# - gcr_image_repo: "gcr.io"
-		# - docker_image_repo: "docker.io"
-		# - quay_image_repo: "quay.io"
-		FIRST_PART=$(echo ${image} | awk -F"/" '{print $1}')
-		if [ "${FIRST_PART}" = "registry.k8s.io" ] ||
-		   [ "${FIRST_PART}" = "gcr.io" ] ||
-		   [ "${FIRST_PART}" = "docker.io" ] ||
-		   [ "${FIRST_PART}" = "quay.io" ] ||
-		   [ "${FIRST_PART}" = "${PRIVATE_REGISTRY}" ]; then
-			image=$(echo ${image} | sed s@"${FIRST_PART}/"@@)
-		fi
-		echo "${FILE_NAME}  ${image}" >> ${IMAGE_LIST}
+	  save_image_to_file "${image}" "${IMAGE_LIST}" || exit $?
 	done
 
 	cd ..
-	sudo chown ${USER} ${IMAGE_DIR}/*
-	tar -zcvf ${IMAGE_TAR_FILE}  ./container-images
-	rm -rf ${IMAGE_DIR}
+	sudo chown -R "${USER}" "${IMAGE_DIR}"
+	tar -zcvf "${IMAGE_TAR_FILE}"  ./container-images
+	rm -rf "${IMAGE_DIR}"
 
 	echo ""
 	echo "${IMAGE_TAR_FILE} is created to contain your container images."
 	echo "Please keep this file and bring it to your offline environment."
 }
 
+function restore_image_file() {
+  file_name=$1
+  raw_image=$2
+  if [ -z "${file_name}" ]; then
+    echo -n "Failed to get file_name"
+    exit 1
+  fi
+  if [ -z "${raw_image}" ]; then
+    echo -n "Failed to get raw_image"
+    exit 1
+  fi
+  if [ ! -f "${IMAGE_DIR}/${file_name}" ]; then
+    echo -n "image archive file ${IMAGE_DIR}/${file_name} do NOT exit"
+    exit 1
+  fi
+
+  org_image=$(sudo docker load -i "${IMAGE_DIR}/${file_name}" | head -n1 | awk '{print $3}')
+  if [ -z "${org_image}" ]; then
+    echo "Failed to get org_image for line ${line}"
+    exit 1
+  fi
+
+  image_id=$(sudo docker image inspect "${org_image}" | grep "\"Id\":" | awk -F: '{print $3}'| sed s/'\",'//)
+  if [ -z "${image_id}" ]; then
+    echo "Failed to get image_id for file ${file_name}"
+    exit 1
+  fi
+
+  sudo docker load -i "${IMAGE_DIR}/${file_name}"
+}
+
+function restore_k8s_image_archives() {
+	if [ ! -f "${IMAGE_LIST}" ]; then
+		echo "image info file ${IMAGE_LIST} should exist."
+		exit 1
+	fi
+
+	while read -r line; do
+
+		file_name=$(echo "${line}" | awk '{print $1}')
+		raw_image=$(echo "${line}" | awk '{print $2}')
+
+    restore_image_file "${file_name}" "${raw_image}"
```

Comment:
ditto

Reply:
The whole script is updated with two-spaces indent style.